base_layer/core/src/chain_storage/blockchain_database.rs

```diff
@@ -157,42 +166,44 @@ pub trait BlockchainBackend: Send + Sync {
         deletions: Vec<HashOutput>,
     ) -> Result<HashOutput, ChainStorageError>;
     /// Constructs a merkle proof for the specified merkle mountain range and the given leaf position.
-    fn fetch_mmr_proof(&self, tree: MmrTree, pos: usize) -> Result<MerkleProof, ChainStorageError>;
-    /// Fetches the checkpoint corresponding to the provided height, the checkpoint consist of the list of nodes
-    /// added & deleted for the given Merkle tree.
-    fn fetch_checkpoint(&self, tree: MmrTree, height: u64) -> Result<MerkleCheckPoint, ChainStorageError>;
+    // DELETE? fn fetch_mmr_proof(&self, tree: MmrTree, pos: usize) -> Result<MerkleProof, ChainStorageError>;
     /// Fetches the total merkle mountain range node count upto the specified height.
     fn fetch_mmr_node_count(&self, tree: MmrTree, height: u64) -> Result<u32, ChainStorageError>;
     /// Fetches the leaf node hash and its deletion status for the nth leaf node in the given MMR tree.
-    fn fetch_mmr_node(&self, tree: MmrTree, pos: u32) -> Result<(Hash, bool), ChainStorageError>;
+    // delete, can use function below with count 1 fn fetch_mmr_node(&self, tree: MmrTree, pos: u32) -> Result<(Hash,
+    // bool), ChainStorageError>;
     /// Fetches the set of leaf node hashes and their deletion status' for the nth to nth+count leaf node index in the
     /// given MMR tree.
     fn fetch_mmr_nodes(&self, tree: MmrTree, pos: u32, count: u32) -> Result<Vec<(Hash, bool)>, ChainStorageError>;
     /// Fetches the leaf index of the provided leaf node hash in the given MMR tree.
     fn fetch_mmr_leaf_index(&self, tree: MmrTree, hash: &Hash) -> Result<Option<u32>, ChainStorageError>;
     /// Performs the function F for each orphan block in the orphan pool.
-    fn for_each_orphan<F>(&self, f: F) -> Result<(), ChainStorageError>
-    where
-        Self: Sized,
-        F: FnMut(Result<(HashOutput, Block), ChainStorageError>);
+    // replace with 2 fn below. // fn for_each_orphan<F>(&self, f: F) -> Result<(), ChainStorageError>
+    // where
+    //     Self: Sized,
+    //     F: FnMut(Result<(HashOutput, Block), ChainStorageError>);
+    /// returns a list of orphan block headers that are parents to the named hash
+    fn get_parent_orphan_headers(&self, hash: HashOutput) -> Result<Vec<BlockHeader>, ChainStorageError>;
```

Comment:
Maybe we should keep things consistent and always use `get` or `fetch` - to me, `fetch` implies that some I/O has to happen on each call, whereas `get` doesn't really tell you either way. Since these will likely involve actually fetching the data on each call, I'd go with `fetch`. But I think either is ok.